SU2_CFD/include/numerics/turbulent/turb_sources.hpp

```diff
@@ -711,34 +709,82 @@ class CSourcePieceWise_TurbSST final : public CNumerics {
     Jacobian_i[1][0] = 0.0;
     Jacobian_i[1][1] = 0.0;
 
+    su2double ProdLimConstant;
+    su2double zeta;
+
     /*--- Computation of blended constants for the source terms ---*/
 
     const su2double alfa_blended = F1_i * alfa_1 + (1.0 - F1_i) * alfa_2;
     const su2double beta_blended = F1_i * beta_1 + (1.0 - F1_i) * beta_2;
 
+    SST_ParsedOptions sstParsedOptions = config->GetSSTParsedOptions();
+
     if (dist_i > 1e-10) {
+
+      const su2double VorticityMag = GeometryToolbox::Norm(3, Vorticity_i);
+      
       /*--- Production ---*/
 
       su2double diverg = 0.0;
       for (unsigned short iDim = 0; iDim < nDim; iDim++)
         diverg += PrimVar_Grad_i[iDim + idx.Velocity()][iDim];
 
-      /*--- If using UQ methodolgy, calculate production using perturbed Reynolds stress matrix ---*/
+      su2double P_Base = StrainMag_i;  //Base production term 
 
-      su2double StrainMag = StrainMag_i;
+      /*--- Modifications of SST production terms (uq, V, KL) --*/
 
-      if (using_uq) {
+      /* --------------------------- */
+      if (sstParsedOptions.production == SST_OPTIONS::UNCERTAINTY) {
         ComputePerturbedRSM(nDim, Eig_Val_Comp, uq_permute, uq_delta_b, uq_urlx, PrimVar_Grad_i + idx.Velocity(),
                             Density_i, Eddy_Viscosity_i, ScalarVar_i[0], MeanPerturbedRSM);
-        StrainMag = PerturbedStrainMag(ScalarVar_i[0]);
+        P_Base = PerturbedStrainMag(ScalarVar_i[0]);
+
+      } else if (sstParsedOptions.production == SST_OPTIONS::VORTICITY) {
+        P_Base = VorticityMag; 
+
+      } else if (sstParsedOptions.production == SST_OPTIONS::KL) {
+        P_Base = sqrt(StrainMag_i*VorticityMag);
+      }  
+      /* ------------------------------ */
+
+
+      /* SST-1994 */
+      if (sstParsedOptions.version == SST_OPTIONS::V1994){
+        ProdLimConstant = 20.0; 
+        P_Base = VorticityMag;
+        /* mut = rho*k/max(w,Omega*F_2/a1) = rho*k/zeta */
+      }
+      else {
+        /* SST-2003 */  
+        ProdLimConstant = 10.0;
+        P_Base = StrainMag_i;
```

Comment:
This implementation is incorrect, the production modifications take precedence over the base version.